lib/_base_viewmodels.dart

```diff
@@ -174,29 +181,46 @@ abstract class MultipleFutureViewModel extends _MultiDataSourceViewModel {
 
 abstract class MultipleStreamViewModel extends _MultiDataSourceViewModel {
   Map<String, Stream> get streamsMap;
+  Map<String, StreamData> _streamDataMap;
 
   void _initialiseData() {
-    _errorMap = Map<String, bool>();
-    _dataMap = Map<String, dynamic>();
+    _streamDataMap = Map<String, StreamData>();
   }
 
-  void _runStreams() {
+  void runStreams() {
+    _initialiseData();
+    notifyListeners();
     for (var key in streamsMap.keys) {
-      //TODO: Implement this
+      _streamDataMap[key] = setupStream(streamsMap[key]);
     }
   }
-
-  void onError({String key, error}) {}
-  void onData(String key) {}
 }
 
 abstract class StreamViewModel<T> extends _SingleDataSourceViewModel<T> {
   Stream<T> get stream;
+  // Holds the data, dataReady, hasError, isBusy properties
+  StreamData streamData;
+  get data => streamData.data;
+  get dataReady => streamData.dataReady;
+  get hasError => streamData.hasError;
+  get isBusy => streamData.isBusy;
+
+  //TODO: Integrate read/write for onData, onCancel, onSubscribed, onError and transformData properties
+  //TODO: Figure out a more elegant way of integrating StreamData
+  //TODO: Get this to work with generic stream types again
+  //TODO: Possibly have a default setbusy for overwrittenn lifecycle events
```

Comment:
For now was thinking about just having setters override the lifecycle events on the `StreamData` class. Let me know if you can think of a better way to implement this, possibly with generics again.